auth/src/main/java/com/firebase/ui/auth/ui/phone/PhoneActivity.java

```diff
@@ -14,396 +14,182 @@
-import com.google.android.gms.tasks.OnFailureListener;
-import com.google.android.gms.tasks.OnSuccessListener;
-import com.google.firebase.FirebaseException;
-import com.google.firebase.auth.AuthResult;
+import com.firebase.ui.auth.viewmodel.ResourceObserver;
+import com.firebase.ui.auth.viewmodel.phone.PhoneProviderResponseHandler;
 import com.google.firebase.auth.FirebaseAuthException;
-import com.google.firebase.auth.FirebaseAuthInvalidCredentialsException;
-import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.auth.PhoneAuthCredential;
 import com.google.firebase.auth.PhoneAuthProvider;
 
-import java.util.concurrent.TimeUnit;
-
 /**
  * Activity to control the entire phone verification flow. Plays host to {@link
- * VerifyPhoneNumberFragment} and {@link SubmitConfirmationCodeFragment}
+ * CheckPhoneNumberFragment} and {@link SubmitConfirmationCodeFragment}
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 public class PhoneActivity extends AppCompatBase {
-    private enum VerificationState {
-        VERIFICATION_NOT_STARTED, VERIFICATION_STARTED, VERIFIED
-    }
-
-    private static final String PHONE_VERIFICATION_LOG_TAG = "PhoneVerification";
-
-    private static final long SHORT_DELAY_MILLIS = 750;
-    @VisibleForTesting static final long AUTO_RETRIEVAL_TIMEOUT_MILLIS = 120000;
-
-    private static final String KEY_VERIFICATION_PHONE = "KEY_VERIFICATION_PHONE";
-    private static final String KEY_STATE = "KEY_STATE";
-
-    private AlertDialog mAlertDialog;
-    @VisibleForTesting CompletableProgressDialog mProgressDialog;
-    private Handler mHandler;
-    private String mPhoneNumber;
-    private String mVerificationId;
-    private Boolean mIsDestroyed = false;
-    private PhoneAuthProvider.ForceResendingToken mForceResendingToken;
-    private VerificationState mVerificationState;
-
-    public static Intent createIntent(Context context, FlowParameters flowParams, Bundle params) {
-        return createBaseIntent(context, PhoneActivity.class, flowParams)
-                .putExtra(ExtraConstants.PARAMS, params);
+    public static Intent createIntent(Context context, FlowParameters params, Bundle args) {
+        return createBaseIntent(context, PhoneActivity.class, params)
+                .putExtra(ExtraConstants.PARAMS, args);
     }
 
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.fui_activity_register_phone);
 
-        mHandler = new Handler();
-        mVerificationState = VerificationState.VERIFICATION_NOT_STARTED;
-        if (savedInstanceState != null && !savedInstanceState.isEmpty()) {
-            mPhoneNumber = savedInstanceState.getString(KEY_VERIFICATION_PHONE);
+        final PhoneProviderResponseHandler handler =
+                ViewModelProviders.of(this).get(PhoneProviderResponseHandler.class);
+        handler.init(getFlowParams());
```

Comment:
nit: name this something more specific than `handler`?  When it's used inside of `verificationHandler` it gets a bit confusing.

Reply:
Renamed `verificationHandler` to `phoneVerifier`.